src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -6,37 +6,44 @@
  *          Tonmoy Saikia, Sergey Lisitsyn, Matt Aasted, Sanuj Sharma
  */
 
-#include <shogun/lib/common.h>
-#include <shogun/kernel/GaussianKernel.h>
-#include <shogun/features/DotFeatures.h>
 #include <shogun/distance/EuclideanDistance.h>
+#include <shogun/features/DotFeatures.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/lib/auto_initialiser.h>
+#include <shogun/lib/common.h>
 #include <shogun/mathematics/Math.h>
 
 using namespace shogun;
 
 GaussianKernel::GaussianKernel() : ShiftInvariantKernel()
 {
-	register_params();
+	set_cache_size(10);
+
+	auto dist=std::make_shared<EuclideanDistance>();
+	dist->set_disable_sqrt(true);
+	m_distance=dist;
+
+	SG_ADD(
+	    &m_log_width, "log_width", "Kernel width in log domain",
+	    ParameterProperties::AUTO | ParameterProperties::HYPER |
+	        ParameterProperties::GRADIENT,
+	    std::make_shared<params::GaussianWidthAutoInit>(*this, 1.0));
```

Comment:
@karlnapf if for some reason it is not possible to compute the median heuristic it defaults to 1, which is the current default

Reply:
1 for bandwidth or for log_width?

Reply:
but yeah it is good. Not sure why it wouldnt work?

Reply:
see other comment I made

Reply:
1 for bandwidth, which is the current default. It might not work if the features are not dense (there is a check for it in the auto initialiser). But I also don't know if the kernel works with non dense features?

Reply:
not sure, but it doesnt make sense as the euclidean distance doesnt mean much in high dimensions

Reply:
I guess this is more of a software completeness thing. If the algorithm doesn't work because of the feature type then it should be handled somewhere else (by the user or the library)

Reply:
agree